### PR TITLE
add node-red

### PR DESCRIPTION
**Pull request by tprelog**
https://nodered.org/

### CODE REVIEW

File: node-red.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "node-red",
+  "release": "11.3-RELEASE",
+  "artifact": "https://github.com/tprelog/iocage-node-red.git",
+  "official": "false",
+  "properties": { 
+    "allow_raw_sockets":"1",
+    "boot": "on",
```
Comment (miwi-fbsd) [inline node-red.json]: Please leave it optional

File: node-red.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "bpf": "on",
```
Comment (miwi-fbsd) [inline node-red.json]: Please remove it automated now.

File: node-red.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "dhcp": "1",
+    "vnet": "yes"
```
Comment (miwi-fbsd) [inline node-red.json]: Please remove it automated now.

**Comment by miwi-fbsd** [comment]
This one looks like the same https://github.com/ix-plugin-hub/iocage-plugin-index/pull/12

**Comment by tprelog** [comment]
That is for 11.2-RELEASE and mine is for 11.3-RELEASE

Also I provide [rc script](https://github.com/tprelog/iocage-node-red/blob/master/overlay/usr/local/etc/rc.d/nodered) with mine.

**Comment by miwi-fbsd** [comment]
> That is for 11.2-RELEASE and mine is for 11.3-RELEASE
> 
When no branch exist it works on both sides, because its using master then as default.

> Also I provide [rc script](https://github.com/tprelog/iocage-node-red/blob/master/overlay/usr/local/etc/rc.d/nodered) with mine.

Also your code base looks a bit better. I will look in a bit.

**Comment by tprelog** [comment]
I updated rc script (11.3-RELEASE) to hopefully address this [request for using max-old-storage-space](https://github.com/tprelog/iocage-node-red/issues/2).

Also I hate to make a change that could break something for people already using this rc script but I am considering such a change. **Can you please give advice?** Here is the situation

Currently in [`/usr/local/etc/rc.d/nodered`](https://github.com/tprelog/iocage-node-red/blob/11.3-RELEASE/overlay/usr/local/etc/rc.d/nodered) have sysrc var `nodered_dir` set to `/usr/home/nodered` by default. This makes sense because `nodered` is also the default user.

I am considering to change the `nodered_dir` default setting to `/var/db/nodered`. I feel this is a more logical location considering someone may wish to change the default user. 

Again this is only to consider changing the _default_ nodered data directory. 

The default option can always be customized
```bash
sysrc nodered_dir=/path/to/directory
```

**Comment by tprelog** [comment]
Is there anything else I need to do for this?

**Comment by kmoore134** [comment]
Ran through a series of tests on it this morning, appears to be in working order now. Thanks for your contribution!

**Comment by tprelog** [comment]
Thanks @kmoore134!

I understand my contributions are in the master branch until they get a final approval.

I'm curious how long does it typically take before something is approved to be merged to the 11.3-RELEASE branch?

Thanks again

**Comment by kmoore134** [comment]
I'm in the process of merging it back to 11.3-RELEASE now, since it appears functional on 11.3-RC1 already.

**Comment by tprelog** [comment]
Wow sweet! Hopefully you get a chance to try the others as well! They were all made against 11.3-RC1

Cheers

**Comment by kmoore134** [comment]
PR opened to merge back to 11.3:

https://github.com/ix-plugin-hub/iocage-plugin-index/pull/23

Martin is going to do a CFT soon and once that happens this will get merged back.

**Pull request merged**